asm/asm8080/asm8080.g4

```diff
@@ -264,9 +264,16 @@ NAME
 
 
 NUMBER
-   : '$'? [0-9a-fA-F] + ('H' | 'h')?
+   : '$'? (DECIMAL | HEXADECIMAL)
    ;
 
+DECIMAL
+   : [0-9]+
+   ;
+
+HEXADECIMAL
+   : [0-9a-fA-F]+ ('H' | 'h')
+   ;
```

Comment:
It's better to use [`fragment`](https://stackoverflow.com/a/6487639/1046374) modifier for these rules because they are used in another lexer tule (`NUMBER`)

```suggestion
fragment DECIMAL
   : [0-9]+
   ;

fragment HEXADECIMAL
   : [0-9a-fA-F]+ ('H' | 'h')
   ;
```